components/scream/src/physics/rrtmgp/rrtmgp_utils.hpp

```diff
@@ -43,6 +46,27 @@ namespace scream {
                 return ( (nstep == 0) || (nstep % irad == 0) );
             }
         }
+
+
+        // Verify that array only contains values within valid range, and if not
+        // report min and max of array
+        template <class T> bool check_range(T x, Real xmin, Real xmax, std::string msg) {
+            bool pass = true;
+            auto _xmin = minval(x);
+            auto _xmax = maxval(x);
+            if (_xmin < xmin or _xmax > xmax) {
+                pass = false;
+                std::cout 
```

Comment:
MInor (if you have time and agree): you could add an optional arg to the signature, which can be used to redirect output somewhere else:
```
template<typename T>
bool check_range (T x, Real xmin, Real xmax, const std::string&, const std::ostream& out = std::cout);
```
The default behavior stays the same (output to screen), but now one can do
```
std::stringstream ss;
check_range (my_val, my_min, my_max, my_msg, ss);

write_to_file (ss);
```

Reply:
Oh that's a great idea, thanks for the suggestion! I'll implement, I want to add a few additional checks anyways.